mmpd-lib/src/midi.rs

```diff
@@ -119,8 +121,121 @@ fn parse_message(bytes: &[u8]) -> Option<MidiMessage> {
     }
 }
 
+/// Parses a string describing a note into MIDI note number(s)
+///
+/// key_str should be in the format "<note name>[accidental][octave]" where angle brackets denote
+/// required content, and square brackets optional content.
+///
+/// Note name is a letter in the range A-G (case-independent)
+/// accidental is either 'b' or '#' and is case sensitive
+/// octave is a single digit decimal number with or without `-` prefix
+///
+/// If no octave is specified, returns a `Vec` with all MIDI note numbers matching that note
+/// If an octave is specified and the resulting note number falls within the MIDI note number
+/// boundaries, a `Vec` with just that number is returned.
+///
+/// An empty `Vec` is returned in the following circumstances:
+/// - The format didn't match that what was described
+/// - The calculated note is out of range of valid MIDI notes
+///
+/// ## Examples
+/// ```
+/// use mmpd_lib::midi::parse_keys_from_str;
+///
+/// let parsed_note = parse_keys_from_str("C3");
+/// assert_eq!(parsed_note, vec![48]);
+///
+/// let parsed_note = parse_keys_from_str("F#4");
+/// assert_eq!(parsed_note, vec![66]);
+///
+/// let parsed_note = parse_keys_from_str("Gb4");
+/// assert_eq!(parsed_note, vec![66]);
+///
+/// let parsed_notes = parse_keys_from_str("C");
+/// assert_eq!(
+///     parsed_notes,
+///     vec![0, 12, 24, 36, 48, 60, 72, 84, 96, 108, 120]
+/// )
+/// ```
+pub fn parse_keys_from_str(key_str: &str) -> Vec<u8> {
+    const NOTES_PER_OCTAVE: i16 = 12;
+    const VALID_NOTES: Range<i16> = 0..128;
+
+    // Breaking apart the regular expression:
+    // ^              - Start of input
+    // ([A-Ga-g])     - Note name capture group: single char that's in range A-G or a-g, required
+    // ([b#]{1,100})? - Accidental capture group: single char that's either 'b' or '#', optional
```

Comment:
"single char" is not correct here, update comment
```suggestion
    // ([b#]{1,100})? - Accidental capture group: one or more chars that are either 'b' or '#', optional
```